### ISSUE TITLE
`useActionData` fails to display data from nested routes on Form submit in Remix@v2.3.0

### ISSUE THREAD

**Issue by tatianainama** [remix-run/remix#8291]
### Reproduction

In Remix version 2.3.0, useActionData does not display submit results from nested routes, contrasting with the behavior in previous version 2.2.0.

-----

1. Go to [StackBlitz Remix@v2.2.0](https://stackblitz.com/edit/remix-run-remix-i3yvp8?file=package.json) 
2. Submit form
3. See submit result in both routes
4. Go to [StackBlitz Remix@v2.3.0](https://stackblitz.com/edit/remix-run-remix-ertyha?file=package.json) 
5. Submit form
6. See submit result only in nested route

<details><summary>v2.2.0 video reproduction</summary>

https://github.com/remix-run/remix/assets/8246284/b4e0d5c1-ecba-441a-8976-37d15603bbed

</details> 

<details><summary>v2.3.0 video reproduction</summary>

https://github.com/remix-run/remix/assets/8246284/e2f4bebb-afa9-474d-be93-f69606354e06

</details> 

-----

No relevant information regarding this change was found in the [changelog for v2.3.0](https://github.com/remix-run/remix/blob/main/CHANGELOG.md#v230). This behavior may be an unintended side-effect or an undocumented bug fix.

### System Info

```shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.18.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.4.2 - /usr/local/bin/npm
    pnpm: 8.10.5 - /usr/local/bin/pnpm
  npmPackages:
    @remix-run/css-bundle: * => 2.4.0 
    @remix-run/dev: 2.3.0 => 2.3.0 
    @remix-run/node: 2.3.0 => 2.3.0 
    @remix-run/react: 2.3.0 => 2.3.0 
    @remix-run/serve: 2.3.0 => 2.3.0
 
```

### Used Package Manager

npm

### Expected Behavior

`useActionData` should display submit results from both main and nested routes.

### Actual Behavior

`useActionData` only displays submit results from main routes.

**Comment by jodygeraldo** [comment]
[this is intended behavior](https://github.com/remix-run/react-router/issues/11022#issuecomment-1828387537)

**Comment by brophdawg11** [comment]
As @jodygeraldo mentioned, this is the correct behavior as was part of a bug fix in 2.3.0 (click through their link for the explanation).

> No relevant information regarding this change was found in the [changelog for v2.3.0](https://github.com/remix-run/remix/blob/main/CHANGELOG.md#v230)

This is still a bit of a tricky relationship we're trying to figure out the best way to document, but effectively all of the code that manages Remix Data APIs (loaders, actions, fetchers) is part of React Router and lives in the `@remix-run/router` package, which has it's own releases and release notes.  

Right now, we've settled on Linking to the new React Router versions for each new Remix release in the [Updated Dependencies](https://github.com/remix-run/remix/blob/main/CHANGELOG.md#updated-dependencies-2) section of the Remix release notes.  Those links take you to the React Router release notes which [include this bug fix](https://github.com/remix-run/react-router/blob/main/CHANGELOG.md#patch-changes-3).

Generally because _so_ much of Remix is really just React Router under the hood, I would recommend clicking through to read those on new Remix releases.
